control-plane/api-gateway/binding/result.go

```diff
@@ -373,14 +373,42 @@ var (
 	// to the GatewayConditionReason given in the spec.
 	errGatewayUnsupportedAddress = errors.New("gateway does not support specifying addresses")
 	errGatewayListenersNotValid  = errors.New("one or more listeners are invalid")
+	errGatewayPending_Pods       = errors.New("gateway pods are still being scheduled")
+	errGatewayPending_Consul     = errors.New("gateway configuration is not yet synced to Consul")
 )
 
 // gatewayValidationResult contains the result of internally validating a gateway.
 // An error set on any of its members corresponds to an error condition on the corresponding
 // status type.
 type gatewayValidationResult struct {
-	acceptedErr error
-	// TODO: programmed
+	acceptedErr   error
+	programmedErr error
+}
+
+// programmedCondition returns a condition for the Programmed status type.
+func (l gatewayValidationResult) programmedCondition(generation int64) metav1.Condition {
+	now := metav1.Now()
+
+	switch l.programmedErr {
+	case errGatewayPending_Pods, errGatewayPending_Consul:
+		return metav1.Condition{
+			Type:               "Programmed",
```

Comment:
do we have this type/reason in either the k8s deps or the consul api dep?

Reply:
Ah, you're right, I didn't realize that the spec now actually [defined these constants](https://github.com/kubernetes-sigs/gateway-api/blob/2dfd20ecb1d829183bb7f5137f7f991feebeb779/apis/v1beta1/gateway_types.go#L549), we should use them. You fine waiting for a refactor pass or need that to go in pre-merge?

Reply:
Fun enough though, due to the typed constants, using any of those values defined in the spec requires casting back to a string, just like we have to do the weird casting elsewhere while using the spec 😞